src/Payments/Payment.php

```diff
@@ -219,243 +221,258 @@ final class Payment extends DateAwareEntity {
-            $value,
+            $paymentAmount,
             $date,
             $paymentMethod,
-            $paymentCurrency,
-            $cashboxId,
             $tags
         );
     }
 
     /**
-     * @param InvoiceId $invoice
-     * @param float $value
-     * @param \DateTimeInterface|null $date
+     * @param InvoiceId $invoiceId
+     * @param PaymentAmount $paymentAmount
+     * @param \DateTime $date
      * @param PaymentMethod|null $paymentMethod
-     * @param PaymentCurrency|null $paymentCurrency
-     * @param PaymentCashboxId|null $cashboxId
      * @param TagIds|null $tags
      * @return Payment
      */
     public static function forInvoiceOfId(
         InvoiceId $invoiceId,
-        $value,
-        \DateTimeInterface $date = null,
-        PaymentMethod $paymentMethod = null,
-        PaymentCurrency $paymentCurrency = null,
-        PaymentCashboxId $cashboxId = null,
+        PaymentAmount $paymentAmount,
+        \DateTime $date,
+        PaymentMethod $paymentMethod,
         TagIds $tags = null
-    ) {
+    ): Payment {
         return new self(
             'invoice',
             $invoiceId,
-            $value,
+            $paymentAmount,
             $date,
             $paymentMethod,
-            $paymentCurrency,
-            $cashboxId,
             $tags
         );
     }
 
     /**
-     * @param Invoice $invoice
-     * @param float $value
-     * @param \DateTimeInterface|null $date
-     * @param PaymentMethod|null $paymentMethod
-     * @param PaymentCurrency|null $paymentCurrency
-     * @param PaymentCashboxId|null $cashboxId
+     * @param ExpenseId $expenseId
+     * @param PaymentAmount $paymentAmount
+     * @param \DateTime $date
+     * @param PaymentMethod $paymentMethod
      * @param TagIds|null $tags
      * @return Payment
      */
     public static function forExpenseOfId(
         ExpenseId $expenseId,
-        $value,
-        \DateTimeInterface $date = null,
-        PaymentMethod $paymentMethod = null,
-        PaymentCurrency $paymentCurrency = null,
-        PaymentCashboxId $cashboxId = null,
+        PaymentAmount $paymentAmount,
+        \DateTime $date,
+        PaymentMethod $paymentMethod,
         TagIds $tags = null
-    ) {
+    ): Payment {
         return new self(
             'expense',
             $expenseId,
-            $value,
+            $paymentAmount,
             $date,
             $paymentMethod,
-            $paymentCurrency,
-            $cashboxId,
             $tags
         );
     }
 
     /**
-     * @return null|\Webit\WFirmaSDK\Entity\EntityId|PaymentId
+     * @return null|PaymentId
      */
-    public function id()
+    public function id(): ?PaymentId
     {
         return PaymentId::create($this->plainId());
     }
 
     /**
-     * @return float
+     * @return \DateTime
      */
-    public function value()
+    public function date(): \DateTime
     {
-        return $this->value;
+        return $this->date->setTime(0, 0);
```

Comment:
Możemy zmienić datę otrzymaną od użytkownika. Idealnie byłoby trzymać DateTimeImmutable.

Reply:
Generalnie możemy pominąć ustawianie `setTime`. W tym kontekście i tak nie ma to znaczenia (API zwraca tylko datę).